feature/savings/src/commonMain/kotlin/org/mifos/mobile/feature/savings/savingsAccountWithdraw/SavingsAccountWithdrawViewModel.kt

```diff
@@ -12,28 +12,40 @@ package org.mifos.mobile.feature.savings.savingsAccountWithdraw
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
-import kotlinx.coroutines.flow.catch
 import kotlinx.coroutines.flow.flatMapLatest
 import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 import org.mifos.mobile.core.common.Constants
-import org.mifos.mobile.core.common.utils.getTodayFormatted
+import org.mifos.mobile.core.common.DataState
+import org.mifos.mobile.core.common.DateHelper
 import org.mifos.mobile.core.data.repository.SavingsAccountRepository
+import org.mifos.mobile.core.data.util.NetworkMonitor
 import org.mifos.mobile.core.model.entity.accounts.savings.SavingsAccountWithdrawPayload
 import org.mifos.mobile.feature.savings.savingsAccountWithdraw.SavingsAccountWithdrawUiState.WithdrawUiReady
-import javax.inject.Inject
 
-@HiltViewModel
-internal class SavingsAccountWithdrawViewModel @Inject constructor(
+internal class SavingsAccountWithdrawViewModel(
     private val savingsAccountRepositoryImp: SavingsAccountRepository,
     savedStateHandle: SavedStateHandle,
+    private val networkMonitor: NetworkMonitor,
 ) : ViewModel() {
 
+    init {
+        viewModelScope.launch {
```

Comment:
Do this outside of the init block. Do it this way 
https://github.com/openMF/mifos-mobile/pull/2775/files#diff-4660f3a70e3e9ee9f31b76e7b05cda295bbb9d897711622b813857897d15c488